### ISSUE TITLE
Keyboard shortcuts

### ISSUE THREAD

**Issue by psychemedia** [innovationOUtside/nbev3devsim#57]
Elements of the widget should be tab-able, so a user can move between them and update the element that currently has the focus, but it might be useful to also support keyboard shortcuts that operate on elements of the widget directly.

The following js fragment perhaps implements something along these lines, but would it conflict with the keydown event captured and used by the notebook UI itself?

```javascript
$(document).ready(function(){
    $('body').keydown(function(){
        console.log('keydown');
        $('#increment').text(parseInt($('#increment').text())+1);   
    });
});
```

The original RobotLab simulator used keyboard shortcuts to support the following operations:

```
 Reset                Shift + Ctrl+Home 
 Show trail 
 Zoom in            Ctrl+NUM PLUS 
 Zoom out         Ctrl+NUM SUB 
 Start clock        Ctrl+Home 
 Pause clock      Ctrl+End 
 Slow down       Ctrl+/ 
 Speed up          Ctrl+NUMMULT
```

**Comment by psychemedia** [comment]
Keymaps are discussed in docs [here](https://jupyter-notebook.readthedocs.io/en/stable/extending/keymaps.html).

**Comment by psychemedia** [comment]
Re: Keyboard focus: one thing that might improve accessibility is to give focus to the simulator run button after downloading some code.

**Comment by psychemedia** [comment]
It would be useful to have shortcuts to:

- run / stop simulator;
- show / hide various elements (chart, display, sensor displays, world canvas) 
- pen up/ pen down; there is also now a limited pen colour selector;
-  arrow keys to move robot; also need to be able to rotate robot (but what angle/degrees step change for each keypress?) (limit to when cursor over robot canvas? Though what if cursor is hidden? May be useful to keyboard control robot in that case?)

Keyboard shortcuts could perhaps also correlate to the [magic CLI switches](https://github.com/innovationOUtside/nbev3devsim/blob/master/nbev3devsim/magics.py) (I really need to document these and/or provide demos; will try to do that over next week or so, not least because I'm losing track of what's possible now...).

Another accessibility thing I'd like to explore more are audible alerts. At this point it may be worth just keeping a note of things that we could sensibly / meaningfully add an audible alert to.

Voice control would also be useful. The [accessibility toolbar](https://github.com/uclixnjupyternbaccessibility/accessibility_toolbar) has code for [voice control ](https://github.com/uclixnjupyternbaccessibility/accessibility_toolbar/tree/master/voice_control) in that extension using *annyang*. Being able to say "RunSim" and "StopSim" for the simulator would be useful.

**Comment by psychemedia** [comment]
Shortcuts now available.
